nixos/modules/services/web-apps/mealie.nix

```diff
@@ -0,0 +1,145 @@
+      type = lib.types.str;
+      default = "http";
+      description = "Protocol to use to serve the service";
+    };
+
+    logLevel = lib.mkOption {
+      # Taken from https://docs.python.org/3/library/logging.html#logging-levels
+      type = lib.types.enum [ "NOTSET" "DEBUG" "INFO" "WARNING" "ERROR" "CRITICAL" ];
+      default = "INFO";
+      description = "Log level to use for this service";
+    };
+
+    extraConfig = lib.mkOption {
+      type = lib.types.attrs;
+      default = {};
+      description = lib.mdDoc ''
+        Some extra configuration for the service.
+        See [the mealie documentation](https://nightly.mealie.io/documentation/getting-started/installation/backend-config/) for available options and default values.
+      '';
+      example = {
+        ALLOW_SIGNUP = "false";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.mealie.extraConfig = {
+      DEFAULT_GROUP = lib.mkDefault "Home";
+      DEFAULT_EMAIL = lib.mkDefault "changeme@example.com";
+      DEFAULT_PASSWORD = lib.mkDefault "MyPassword";
+      ALLOW_SIGNUP = lib.mkDefault "true";
+
+      API_DOCS = lib.mkDefault "False";
+
+      TOKEN_TIME = lib.mkDefault 24;
+      TZ = lib.mkDefault "UTC";
+
+      DB_ENGINE = lib.mkDefault "sqlite";
+      SECURITY_MAX_LOGIN_ATTEMPTS = lib.mkDefault "5";
+      SECURITY_USER_LOCKOUT_TIME = lib.mkDefault "24";
+
+      POSTGRES_USER = lib.mkDefault "mealie";
+      POSTGRES_PASSWORD = lib.mkDefault "mealie";
+      POSTGRES_SERVER = lib.mkDefault "postgres";
+      POSTGRES_PORT = lib.mkDefault 5432;
+      POSTGRES_DB = lib.mkDefault "mealie";
+
+      SMTP_PORT = lib.mkDefault 587;
+      SMTP_FROM_NAME = lib.mkDefault "mealie";
+      SMTP_AUTH_STRATEGY = lib.mkDefault "TLS";
+
+      WEB_GUNICORN = lib.mkDefault "false";
+      WORKERS_PER_CORE = lib.mkDefault 1;
+      MAX_WORKERS = lib.mkDefault 1;
+      WEB_CONCURRENCY = lib.mkDefault 1;
+
+      LDAP_AUTH_ENABLED = lib.mkDefault "False";
+      LDAP_TLS_INSECURE = lib.mkDefault "False";
+      LDAP_ENABLE_STARTTLS = lib.mkDefault "False";
+      LDAP_ID_ATTRIBUTE = lib.mkDefault "uid";
+      LDAP_NAME_ATTRIBUTE = lib.mkDefault "name";
+      LDAP_MAIL_ATTRIBUTE = lib.mkDefault "mail";
+
+      THEME_LIGHT_PRIMARY = lib.mkDefault "#E58325";
+      THEME_LIGHT_ACCENT = lib.mkDefault "#007A99";
+      THEME_LIGHT_SECONDARY = lib.mkDefault "#973542";
+      THEME_LIGHT_SUCCESS = lib.mkDefault "#43A047";
+      THEME_LIGHT_INFO = lib.mkDefault "#1976D2";
+      THEME_LIGHT_WARNING = lib.mkDefault "#FF6D00";
+      THEME_LIGHT_ERROR = lib.mkDefault "#EF5350";
+      THEME_DARK_PRIMARY = lib.mkDefault "#E58325";
+      THEME_DARK_ACCENT = lib.mkDefault "#007A99";
+      THEME_DARK_SECONDARY = lib.mkDefault "#973542";
+      THEME_DARK_SUCCESS = lib.mkDefault "#43A047";
+      THEME_DARK_INFO = lib.mkDefault "#1976D2";
+      THEME_DARK_WARNING = lib.mkDefault "#FF6D00";
+      THEME_DARK_ERROR = lib.mkDefault "#EF5350";
```

Comment:
That _way_ too many things, especially since they have default values in the app.

```suggestion
```